### ISSUE TITLE
Rewrite App.js from class to function component with hooks

### ISSUE THREAD

**Issue by draperunner** [entur/shamash#20]
App.js is a class component using deprecated lifecycle methods (`componentWillMount`). It would be nice to rewrite it as a function component with React hooks.

**Comment by echeveria** [comment]
If a have time this weekend, i will try to fix it.

**Comment by echeveria** [comment]
A have a question. What is this, same functionality but different names:
```
 onEditQuery(query) {
    this.setState({
      parameters: {
        ...this.state.parameters,
        query
      }
    });
    this.updateURL();
  }

  onEditVariables(variables) {
    this.setState({
      parameters: {
        ...this.state.parameters,
        variables
      }
    });
    this.updateURL();
  }

  onEditOperationName(operationName) {
    this.setState({
      parameters: {
        ...this.state.parameters,
        operationName
      }
    });
    this.updateURL();
  }
```

**Comment by draperunner** [comment]
Just a case of non-DRY (wet?) programming. There's a slight difference: the name of the parameter. I guess we could combine them into something like

```
onEditParameter(key, value) {
    this.setState({
      parameters: {
        ...this.state.parameters,
        [key]: value
      }
    });
    this.updateURL();
}
```

**Comment by echeveria** [comment]
Ok, try to merge, but may be have conflict.

**Comment by draperunner** [comment]
Fixed by #25
